tests/jobs/test_api.py

```diff
@@ -64,3 +64,81 @@ def test_list_jobs_by_name(jobs_api):
     assert len(res) == 2
     assert res[0]['settings']['name'] == test_job_name
     assert res[1]['settings']['name'] == test_job_name
+
+
+@provide_conf
+def test_delete_job():
```

Comment:
Haven't done this in a long time, but something like this should work (same below, but not sure if that's even more pythonic way to do it)?
```
def provide_mock_client(fn):
  def inner(*args, **kwargs):
     with mock.patch('databricks_cli.sdk.ApiClient') as api_client_mock:
        fn(api_client_mock, *args, **kwargs)
  return inner

@provide_conf
@provide_mock_client
def test_delete_job(api_client_mock):
        api = JobsApi(api_client_mock)
        api.delete_job('1')
        api_client_mock.perform_query.assert_called_with(
            'POST', '/jobs/delete', data={'job_id': '1'}, headers=None, version=None
        )
        api.delete_job('1', version='3.0')
        api_client_mock.perform_query.assert_called_with(
            'POST', '/jobs/delete', data={'job_id': '1'}, headers=None, version='3.0'
        )
```